src/mongo/db/pipeline/expression.h

```diff
@@ -443,13 +446,22 @@ class ExpressionFromAccumulator
         return accum.getValue(false);
     }
 
-    bool isAssociativeAndCommutative() const final {
+    bool isAssociative() const final {
         // Return false if a single argument is given to avoid a single array argument being treated
         // as an array instead of as a list of arguments.
         if (this->vpOperand.size() == 1) {
             return false;
         }
-        return Accumulator().isAssociativeAndCommutative();
+        return Accumulator().isAssociative();
+    }
+
+    bool isCommutative() const final {
```

Comment:
I think this one can always return `Accumulator().isCommutative()`. The operand size check was intended to prevent coalescing in the case described in my comment about adding the `nary->isAssociative()` check above. I don't see anything that can go wrong if `isCommutative()` returns true in that scenario (or any other scenarios).